### DISCUSSION TITLE
Memork leak

### DISCUSSION THREAD

**Discussion by ohuo1215** [coinfoundry/miningcore#1162] [Q&A] [answered]
As MiningCore runs, its memory usage increases until the service stops running when its memory usage reaches 30%  (bus error)
![ad92f6e981ec6c0ab0219d93da7833e](https://user-images.githubusercontent.com/52817197/153572741-66e16f29-3803-421b-9b37-c8e1d561101a.png)
.

**Comment by oliverw** [comment]
I don't see any indication of a memory leak from the information you provided. Ethereum is taxing on the system because the pool may have to keep the current DAG plus the previous one in memory at some point.

**Reply by ohuo1215** [reply]
I'm really sorry that I forgot to place the relevant pictures
![7410b25aca19605c3ecf52de75053fa](https://user-images.githubusercontent.com/52817197/153607903-e4d47a44-356f-4553-9153-04ccf1091b4f.png)

**Comment by oliverw** [comment, marked as answer]
This is still not a convincing indicator of a memory leak. Memory usage going up happens all the time when a garbage collector is involved as its the case with .NET. How much memory has the system you are running on? Is it a container a VM, bare metal?

**Reply by ohuo1215** [reply]
sorry
I am using a purchased virtual machine (2 cores 8 GiB).  and，I‘m now installing a monitoring tool to see if the memory leak is real

**Reply by oliverw** [reply]
8 GB RAM is way to little ram for ETH.

**Reply by ohuo1215** [reply]
I would like to know what these files are and whether they can be deleted because they occupy too much space of my hard disk 
![4d83061f8c41cd4c33d3ff396e6a75a](https://user-images.githubusercontent.com/52817197/153708545-971377ef-6c8b-45e0-a150-c138f8b3d732.png)

**Reply by oliverw** [reply]
Those are old DAGs. All but the newest two files can be deleted.

**Reply by ohuo1215** [reply]
Thank you very much. Now MiningCore will not stop by accident. The reason why miningCore stops is that the disk space is full.

**Reply by ohuo1215** [reply]
But I have another question, how should I set up the system to automatically clean these expired DAGs.

**Reply by oliverw** [reply]
Set up a cron job on your system that runs the following **inside** your dag directory:

`ls -1tr | head -n -2 | xargs -d '\n' rm -f --`

**Reply by ohuo1215** [reply]
Okay, thank you very much
